test/unit/fixtures.go

```diff
@@ -0,0 +1,59 @@
+package unit
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// TestFixtures manages loading and retrieving test fixtures
+type TestFixtures struct {
+	fixtures map[string]interface{}
+}
+
+// NewTestFixtures creates a new TestFixtures instance
+func NewTestFixtures() *TestFixtures {
+	tf := &TestFixtures{}
+	tf.loadFixtures()
+	return tf
+}
+
+// loadFixtures loads all JSON files in fixtures directory
+func (tf *TestFixtures) loadFixtures() {
```

Comment:
optional: One other potential way to handle this would be using an [embed directive](https://pkg.go.dev/embed) with the embed.FS type. I think the current implementation is perfectly acceptable too, just figured I'd share another potential approach 🙂

Reply:
Appreciate the feedback. I wasn't aware of this option at all, but it seems like a great improvement to implement. This change should enhance performance as the test framework scales with additional tests and components 👍